mod/solr_api/start.php

```diff
@@ -0,0 +1,302 @@
+					LEFT JOIN {$db_prefix}entities e ON ue.guid = e.guid 
+				WHERE e.enabled = 'yes'
+				LIMIT 10 OFFSET {$offset}";
+
+	$users = get_data($query);
+
+	foreach ($users as $user) {
+		$arr[] = array (
+			'guid' => $user->guid,
+			'name' => array('en' => $user->name, 'fr' => $user->name),
+			'username' => $user->username,
+			'email' => $user->email,
+			'type' => $user->type,
+			'date_created' => date("Y-m-d\TH:m:s\Z", $user->time_created),
+			'date_modified' => date("Y-m-d\TH:m:s\Z", $user->time_created),
+			'url' => "{$site_url}profile/{$user->username}"
+		);
+	}
+    return $arr;
+}
+
+
+function get_group_list($offset) {
+	// @TODO use SQL query syntax instead of using function elgg_get_entities()
+	$groups = elgg_get_entities(array(
+		'type' => 'group',
+		'limit' => 10,
+		'offset' => $offset
+	));
+
+	foreach ($groups as $group) {
+
+		if (is_Json($group->name)) {
+			$name_array = json_decode($group->name, true);
+			$name_array['en'] = str_replace('"', '\"', $name_array['en']);
+			$name_array['fr'] = str_replace('"', '\"', $name_array['fr']);
+		} else {
+			$name_array['en'] = $group->name_array;
+			$name_array['fr'] = $group->name;
+		}
+
+		if (is_Json($group->description)) {
+			$description_array = json_decode($group->description, true);
+			$description_array['en'] = str_replace('"', '\"', $description_array['en']);
+			$description_array['fr'] = str_replace('"', '\"', $description_array['fr']);
+		} else {
+			$description_array['en'] = $group->description;
+			$description_array['fr'] = $group->description;
+		}
+
+		$arr[] = array(
+			'guid' => $group->getGUID(),
+			'name' => $name_array,
+			'description' => $description_array,
+			'type' => $group->getType(),
+			'access_id' => $group->access_id,
+			'date_created' => date("Y-m-d\TH:m:s\Z", $group->time_created),
+			'date_modified' => date("Y-m-d\TH:m:s\Z", $group->time_updated),
+			'url' => $group->getURL()
+		);
+	}
+    return $arr;
+}
+
+
+function get_entity_list($type, $subtype, $offset) {
+	$entities = elgg_get_entities(array(
+		'type' => $type,
+		'subtype' => $subtype,
+		'limit' => 10,
+		'offset' => $offset
+	));
+
+	foreach ($entities as $entity) {
+
+
+		if (is_Json($entity->title)) {
+			$title_array = json_decode($entity->title, true);
+			if (!isset($title_array['en']) || !isset($title_array['fr'])) {
+				$title_array['en'] = str_replace('"', '\"', $title_array);
+				$title_array['fr'] = str_replace('"', '\"', $title_array);
+			} else {
+				$title_array['en'] = str_replace('"', '\"', gc_explode_translation($entity->title, 'en'));
+				$title_array['fr'] = str_replace('"', '\"', gc_explode_translation($entity->title, 'fr'));
+			}
+
+		} else {
+			$title_array['en'] = $entity->title;
+			$title_array['fr'] = $entity->title;
+		}
+
+		if (is_Json($entity->description)) {
+			$description_array = json_decode($entity->description, true);
+			if (!isset($description_array['en']) || !isset($description_array['en'])) {
```

Comment:
one of them should be fr, or you can just let gc_explode_translation() handle the logic if you want to use it.